src/libcore/ops/range.rs

```diff
@@ -696,6 +696,28 @@ pub enum Bound<T> {
     Unbounded,
 }
 
+impl<T: Clone> Bound<&T> {
+    /// Map a `Bound<&T>` to a `Bound<T>` by cloning the contents of the bound.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::ops::Bound::*;
+    /// use std::ops::RangeBounds;
+    ///
+    /// assert_eq!((1..12).start_bound(), Included(&1));
+    /// assert_eq!((1..12).start_bound().cloned(), Included(1));
+    /// ```
+    #[unstable(feature = "bound_cloned", issue = 61356)]
+    fn cloned(&self) -> Bound<T> {
+        match self {
+            Bound::Unbounded => Bound::Unbounded,
+            Bound::Included(x) => Bound::Included(x.clone()),
```

Comment:
You're getting bitten by pattern auto-ref here, since you match on `&Self` without putting any `&` in the pattern. It's as if you matched `&Bound::Included(ref x)`, so `x: &&T` and `x.clone()` returns a new `&T`.

Reply:
Man I remember when we had to put the `ref` keyword in.  That defined fighting the borrow checker for me.  Literally that was enough for me to be fed up and head back to C++.